go/saltpack/decrypt.go

```diff
@@ -160,13 +164,22 @@ func (ds *decryptStream) tryVisibleReceivers(hdr *EncryptionHeader, ephemeralKey
 func (ds *decryptStream) tryHiddenReceivers(hdr *EncryptionHeader, ephemeralKey BoxPublicKey) (BoxSecretKey, BoxPrecomputedSharedKey, []byte, int) {
 	secretKeys := ds.ring.GetAllSecretKeys()
 
+	for _, r := range hdr.Receivers {
+		if len(r.ReceiverKID) == 0 {
+			ds.mki.NumAnonReceivers++
+		}
+	}
+
 	for _, secretKey := range secretKeys {
+
 		shared := secretKey.Precompute(ephemeralKey)
 
 		for i, r := range hdr.Receivers {
-			keysRaw, err := shared.Unbox(ds.nonce.ForKeyBox(), r.Keys)
-			if err == nil {
-				return secretKey, shared, keysRaw, i
+			if len(r.ReceiverKID) == 0 {
```

Comment:
The case I'm curious about is where a receiver box is listed as key Foo (which we don't have), but it's _actually_ encrypted for key Bar (which we do have). It looks like we never try key Bar against that box. Super corner case, and I don't really feel strongly either way about what we should do with it.